projects/output-and-error-handling/parseResponse_test.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+	"time"
+)
+
+func TestParseResponse(t *testing.T) {
+
+	tests := map[string]struct {
+		input        http.Response
+		expectedBody string
+		expectedErr  RequestError
+	}{
+		"429":     {input: http.Response{StatusCode: 429, Header: http.Header{"Retry-After": []string{"5"}}, Body: io.NopCloser(bytes.NewBufferString(""))}, expectedBody: "", expectedErr: RequestError{SleepTime: time.Duration(5), ShouldRetry: true, Cause: fmt.Errorf("server is busy, retrying in few seconds")}},
```

Comment:
In this kind of test, I'd typically expect each field to be on its own line to make it easier to skim what the inputs and outputs are, rather than needing to read the details across a whole big line of stuff.